pyMBE.py

```diff
@@ -722,101 +722,110 @@ def create_counterions(self, object_name, cation_name, anion_name, espresso_syst
-                    previous_residue_id = central_bead_id
-                    first_residue = False          
-                else:                    
-                    previous_central_bead_name=self.df[self.df['name']==previous_residue].central_bead.values[0]
-                    new_central_bead_name=self.df[self.df['name']==residue].central_bead.values[0]       
-                    bond = self.search_bond(particle_name1=previous_central_bead_name, 
-                                            particle_name2=new_central_bead_name, 
-                                            hard_check=True, 
-                                            use_default_bond=use_default_bond)                
-                    l0 = self.get_bond_length(particle_name1=previous_central_bead_name, 
-                                              particle_name2=new_central_bead_name, 
-                                              hard_check=True, 
-                                              use_default_bond=use_default_bond)                
-                    residue_position = residue_position+backbone_vector*l0
-                    residues_info = self.create_residue(name=residue, 
-                                                        number_of_residues=1, 
-                                                        espresso_system=espresso_system, 
-                                                        central_bead_position=[residue_position],
-                                                        use_default_bond= use_default_bond, 
-                                                        backbone_vector=backbone_vector)
-                    residue_id = next(iter(residues_info))      
-                    for index in self.df[self.df['residue_id']==residue_id].index:
-                        if not self.check_if_df_cell_has_a_value(index=index,key=('molecule_id','')):
-                            self.df.at[index,'molecule_id'] = molecule_id
+            self.copy_df_entry(name=name,
+                            column_name='molecule_id',
+                            number_of_copies=number_of_molecules)
+            
+            molecules_index = self.np.where(self.df['name']==name)
+            molecule_index_list =list(molecules_index[0])[-number_of_molecules:]
+            used_molecules_id = self.df.molecule_id.dropna().drop_duplicates().tolist()
+            pos_index = 0 
+            for molecule_index in molecule_index_list:        
+                molecule_id = self.assign_molecule_id (name=name,pmb_type='molecule',used_molecules_id=used_molecules_id,molecule_index=molecule_index)
+                for residue in residue_list:
+                    if first_residue:
+                        if list_of_first_residue_positions == None:
+                            residue_position = None
+                        else:
+                            if len(list_of_first_residue_positions) != number_of_molecules:
```

Comment:
This sanity test should be moved to the begging of the function (there is no need to check for each loop iteration)